src/library_wasmfs_opfs.js

```diff
@@ -285,7 +285,11 @@ mergeInto(LibraryManager.library, {
   _wasmfs_opfs_read_access: function(accessID, bufPtr, len, pos) {
     let accessHandle = wasmfsOPFSAccessHandles.get(accessID);
     let data = HEAPU8.subarray(bufPtr, bufPtr + len);
-    return accessHandle.read(data, {at: pos});
+    try {
+      return accessHandle.read(data, {at: pos});
+    } catch (e) {
+      return -1;
+    }
```

Comment:
Hmm, weren't we saying that `errPtr` is the simpler thing, than to overload the return values?

And specifically, overloading it here means we are limited to 2GB, I think.

Reply:
It turns out using negative return values was already the established pattern in this file, but maybe we should update it everywhere to use error pointers. Notably we can use the return value directly here because this is not an async function.

Reply:
Also, there is a TODO in opfs_backend.cpp about using an i64 here, which would eliminate the 2GB limit.

Reply:
Also see https://github.com/emscripten-core/emscripten/pull/17774, which tries to make the existing error handling more uniform and robust.